mapie/multi_label_classification.py

```diff
@@ -0,0 +1,790 @@
+        ):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a classifier with fit,"
+                "predict, and predict_proba methods."
+            )
+        check_is_fitted(est)
+        return estimator, X, y
+
+    def _check_partial_fit_first_call(self) -> bool:
+        """
+        Check that this is the first time partial_fit
+        or fit is called.
+
+        Returns
+        -------
+        bool
+            True if it is the first time, else False.
+        """
+        return not hasattr(self, "risks")
+
+    def _check_bound(self, bound: Optional[str]):
+        """
+        Check the value of the bound.
+
+        Parameters
+        ----------
+        bound : Optional[str]
+            Bound defined in the predict.
+
+        Raises
+        ------
+        AttributeError
+            If bound is not in ["hoeffding", "bernstein", "wsr", None]
+        """
+        if bound not in self.valid_bounds_:
+            raise ValueError(
+                "bound must be in ['hoeffding', 'bernstein', 'wsr', None]"
+            )
+        elif (bound is not None) and (self.method == "crc"):
+            warnings.warn(
+                "WARNING: you are using crc method, hence "
+                + "even if the bound is not None, it won't be"
+                + "taken into account"
+            )
+
+    def _transform_pred_proba(
+        self,
+        y_pred_proba: Union[Sequence[NDArray], NDArray]
+    ) -> NDArray:
+        """If the output of the predict_proba is a list of arrays (output of
+        the ``predict_proba`` of ``MultiOutputClassifier``) we transform it
+        into an array of shape (n_samples, n_classes)
+
+        Parameters
+        ----------
+        y_pred_proba : Union[List, NDArray]
+            Output of the multi-label classifier.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes)
+            Output of the model ready for risk computation.
+        """
+        if isinstance(y_pred_proba, np.ndarray):
+            y_pred_proba_array = y_pred_proba
+        else:
+            y_pred_proba_stacked = np.stack(
+                y_pred_proba,  # type: ignore
+                axis=0
+            )[:, :, 1]
+            y_pred_proba_array = np.moveaxis(y_pred_proba_stacked, 0, -1)
+
+        return np.expand_dims(y_pred_proba_array, axis=2)
+
+    def _compute_risks(self, y_pred_proba: NDArray, y: NDArray) -> NDArray:
+        """
+        Compute the risks
+
+        Parameters
+        ----------
+        y_pred_proba : NDArray of shape (n_samples, n_labels)
+            Predicted probabilities for each label and each observation
+        y : NDArray of shape (n_samples, n_labels)
+            True labels.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_lambdas)
+            Risks for each observation and each value of lambda.
+        """
+        lambdas = np.arange(
```

Comment:
we wanted to set lambdas as a attribute instead of n_lambdas, no?

Reply:
done